Automattic/woocommerce-payments
# Use timezone to check WooPay available countries

Fixes 1839-gh-Automattic/woopay

#### Changes proposed in this Pull Request









Use the timezone and exact time to request for WooPay available countries, preventing all stores with WooPay enabled to request for it at the same time.

#### Testing instructions





* Access [phpMyAdmin](http://localhost:8083/index.php?route=/table/sql&db=wordpress&table=wp_options) and run this query:

```
DELETE FROM `wp_options` WHERE `option_name` = 'woocommerce_woocommerce_payments_woopay_available_countries_last_check'
```

* Comment [these lines](https://github.com/Automattic/woocommerce-payments/blob/218b3e01ceb52c31212120f556e5717a1fd0f9b9/includes/platform-checkout/class-platform-checkout-utilities.php#L172-L174) to ignore test mode.
* Access the product, cart or checkout page.
* Run this query on phpMyAdmin:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woocommerce_woocommerce_payments_woopay_available_countries'
```

* Modify the countries array.
* Reload the page.
* Refresh the query on phpMyAdmin, the countries list should not be updated, meaning the page reload got the data from cache.
* Run this query:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woocommerce_woocommerce_payments_woopay_available_countries_last_check'
```

* The option value should be today date and hour.
* Modify it to the date and time be a date in the past, more than 1 day.
* Reload the page.
* Run this query:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woocommerce_woocommerce_payments_woopay_available_countries'
```

* The value should be the current array of WooPay available countries, meaning the request to update its value was made.
* Run this query:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woocommerce_woocommerce_payments_woopay_available_countries_last_check'
```

* Its value should be the date and time you reloaded the page for the last time.
* Reload the page.
* Refresh the latest query, its value should continue to be the same value.

-------------------


**Post merge**
